go-controller/pkg/ovn/master.go

```diff
@@ -159,20 +159,20 @@ func (oc *Controller) upgradeToSingleSwitchOVNTopology(existingNodeList *kapi.No
 		}
 	}
 
-	nodeSwitches, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading", "--format=csv",
-		"--columns=name", "find", "logical_switch")
-	if err != nil {
-		return fmt.Errorf("failed to get all logical switches for upgrade: stderr: %q, error: %v",
-			stderr, err)
+	nodeSwitches := []nbdb.LogicalSwitch{}
+	if err := oc.nbClient.WhereCache(func(lr *nbdb.LogicalSwitch) bool {
+		return true
+	}).List(&nodeSwitches); err != nil {
+		return err
 	}
 
 	logicalNodes := make(map[string]bool)
-	for _, switchName := range strings.Split(nodeSwitches, "\n") {
+	for _, nodeSwitch := range nodeSwitches {
 		// We are interested only in the join_* switches
-		if !strings.HasPrefix(switchName, "join_") {
+		if !strings.HasPrefix(nodeSwitch.Name, "join_") {
 			continue
 		}
-		nodeName := strings.TrimPrefix(switchName, "join_")
+		nodeName := strings.TrimPrefix(nodeSwitch.Name, "join_")
 		logicalNodes[nodeName] = true
```

Comment:
Or you can do all this within `WhereCache`

Anyway, weird way to find out node names.